web/modules/custom/joinup_search/joinup_search.behat.inc

```diff
@@ -20,9 +21,20 @@ use PHPUnit\Framework\Assert;
  */
 class SearchSubContext extends DrupalSubContextBase implements DrupalSubContextInterface {
 
+  use BrowserCapabilityDetectionTrait;
+  use KeyboardInteractionTrait;
   use TraversingTrait;
   use UtilityTrait;
-  use BrowserCapabilityDetectionTrait;
+
+  /**
+   * Navigates to the search page.
+   *
+   * @Given I am on the search( results)( page)
+   * @When (I )visit the search( results)( page)
```

Comment:
Again, for both of these, some combinations don't make much sense.
`Given I am on the search` without any suffix is also not very accurate since it includes the `the` keyword when it is the keyword `page` that is the noun. The `search` keyword is kind of an abstract word that could be counted as an adjective and is not necessarily perceived as a noun.
I suggest something like 
```
@Given I am on the search( results) page
@When I visit the search( results) page
```

Reply:
Yes please! See https://github.com/ec-europa/joinup-dev/pull/1554 for a more comprehensive solution.